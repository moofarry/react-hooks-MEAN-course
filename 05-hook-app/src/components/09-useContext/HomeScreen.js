import React from "react";
import { UserContext } from "./UserContext";

export const HomeScreen = () => {

  const { user } = React.useContext(UserContext);

  return (
    <div>
      <h1>HomeScreen</h1>
      <hr />

      <pre>
        {JSON.stringify( user,null,3)}
      </pre>
    </div>
  );
};
