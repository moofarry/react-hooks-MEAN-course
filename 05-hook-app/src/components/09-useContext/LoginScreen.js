import React from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {

  const { setUser } = React.useContext(UserContext);

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />

      <button className="btn btn-primary" onClick={() => setUser({ id: 123, name: "jhon" })}>
        Login
      </button>
    </div>
  );
};
