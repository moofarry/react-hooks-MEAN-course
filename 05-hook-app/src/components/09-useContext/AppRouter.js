import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import { AboutScreen } from "./AboutScreen";
import { HomeScreen } from "./HomeScreen";
import { LoginScreen } from "./LoginScreen";
import { NavBar } from "./NavBar";

export const AppRouter = () => {
  return (
    <Router>
      <NavBar />
      <div className="container">
        <Routes>
          <Route path="/" element={<HomeScreen />} />
          <Route path="/about" element={<AboutScreen />} />
          <Route path="/login" element={<LoginScreen />} />

          <Route path="*" element={<HomeScreen />} />
        </Routes>
      </div>
    </Router>
  );
};
