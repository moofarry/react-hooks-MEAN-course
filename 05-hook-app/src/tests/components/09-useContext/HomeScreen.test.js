import React from 'react'
import { mount } from "enzyme";
import { UserContext } from '../../../components/09-useContext/UserContext';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';

describe("test in <HomeScreen/>", () => {

  const user ={
    name: 'jhon',
    email: 'jhon@yopmail.com'
  }
  const wrapper = mount(
    <UserContext.Provider value= {{user}}>
      <HomeScreen/>
    </UserContext.Provider>
  )

  test("should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
