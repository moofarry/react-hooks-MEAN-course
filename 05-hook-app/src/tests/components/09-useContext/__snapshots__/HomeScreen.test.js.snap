// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <HomeScreen/> should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "jhon",
   "email": "jhon@yopmail.com"
}
  </pre>
</div>
`;
